modules/twinklefluff.js

```diff
@@ -395,13 +395,61 @@ Twinkle.fluff.revertToRevision = function revertToRevision(oldrev) {
 		type: 'csrf',
 		format: 'json'
 	};
-	var wikipedia_api = new Morebits.wiki.api('Grabbing data of the earlier revision', query, Twinkle.fluff.callbacks.toRevision);
-	wikipedia_api.params = { rev: oldrev };
+	var wikipedia_api = new Morebits.wiki.api('Grabbing data of the earlier revision', query, Twinkle.fluff.getSummary);
+	wikipedia_api.params = { rev: oldrev, type: 'torev' };
 	wikipedia_api.post();
 };
 
+Twinkle.fluff.getSummary = function getSummary(apiobj) {
+	Twinkle.fluff.apiobj = apiobj;
+	var params = apiobj.params;
+	var type = params.type;
+	if (type === 'vand' || (type === 'norm' && !Twinkle.getPref('offerReasonOnNormalRevert'))) {
+		Twinkle.fluff.callbacks.main('noSummary');
+	} else {
+		Twinkle.fluff.simpleWindow = new Morebits.simpleWindow(600, 350);
+		Twinkle.fluff.simpleWindow.setTitle('Enter reversion edit summary');
+		Twinkle.fluff.simpleWindow.setScriptName('Twinkle');
+		Twinkle.fluff.simpleWindow.addFooterLink('Revert and rollback prefs', 'WP:TW/PREF#fluff');
+		Twinkle.fluff.simpleWindow.addFooterLink('Twinkle help', 'WP:TW/DOC#fluff');
+		var form = new Morebits.quickForm(type === 'torev' ? Twinkle.fluff.callbacks.toRevision : Twinkle.fluff.callbacks.main);
+		if (Twinkle.getPref('customReversionSummaryList').length) {
+			form.append({
+				type: 'radio',
+				name: 'summaryList',
+				list: Twinkle.getPref('customReversionSummaryList'),
+				event: function(event) {
+					var esForm = event.target.form;
+					esForm.summary.value = this.value;
+					event.stopPropagation();
+				}
+			});
+		}
+		form.append({
+			id: 'editSummaryInput',
+			type: 'input',
```

Comment:
What about a `select` option?  Less space, simpler interface as the list gets long?  An extra click, but maybe that's not a big deal, especially when compared to the current manual entry.